leech.py

```diff
@@ -1,79 +1,99 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
-import argparse
-import sys
-import json
+import click
+from click_default_group import DefaultGroup
```

Comment:
The [click docs](http://click.pocoo.org/6/commands/#group-invocation-without-command) imply that you can just do `@click.group(invoke_without_command=True)`, so this seems like an unnecessary dependency?

Reply:
invoke_without_command does something subtly different. The contents of the group are normally executed as a way to set up common context. If invoke_without_command=True, the group actions will be run even if there is not subcommand.

In this case though, the desired behavior is that if no subcommand is specified, we should act as if a specific subcommand (download) was in fact specified.

It is possible, of course, to check if a subcommand is specified, and artificially invoke a specific subcommand (e.g. like the example you showed). This can be a little messy though (since you have to make sure all the right args end up in the right places) and is almost line for line what click_default_group does.